jerry-core/jerry.c

```diff
@@ -1845,81 +1828,112 @@ snapshot_add_compiled_code (ecma_compiled_code_t *compiled_code_p, /**< compiled
   if ((snapshot_buffer_write_offset >> MEM_ALIGNMENT_LOG) > 0xffffu)
   {
     snapshot_error_occured = true;
-    return;
+    return 0;
   }
 
-  snapshot_last_compiled_code_offset = snapshot_buffer_write_offset;
+  uint16_t start_offset = (uint16_t) (snapshot_buffer_write_offset >> MEM_ALIGNMENT_LOG);
+  ecma_compiled_code_t *copied_compiled_code_p;
 
-  compiled_code_map_entry_t *new_entry;
-  new_entry = (compiled_code_map_entry_t *) mem_heap_alloc_block (sizeof (compiled_code_map_entry_t));
+  copied_compiled_code_p = (ecma_compiled_code_t *) (snapshot_buffer_p + snapshot_buffer_write_offset);
```

Comment:
Would break the 120 character maximum line length. Other kind of line breaking is also not allowed so this is our only option currently.